commit aa6fd026ecd2a53c7e1f62ef0bbf5dd80b4671dc
Author: Christian Lamparter <chunkeey@web.de>
Date:   Sat Mar 21 23:11:49 2009 +0100

    ar9170: update Makefile, Kconfig and MAINTAINERS
    
    This patch update all necessary Makefile and Kconfig files.
    
    Now you can then enable ar9170 in the kernel configuration under:
    Device Drivers --->
    	[*] Network device support --->
    		Wireless LAN --->
    			[*] Wireless LAN (IEEE 802.11)
    			<M>   Atheros AR9170 support
    			<M>     Atheros AR9170 USB support
    
    Signed-off-by: Christian Lamparter <chunkeey@web.de>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/MAINTAINERS b/MAINTAINERS
index 3c8aa4d..3d0caf3 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -765,6 +765,14 @@ L:	linux-wireless@vger.kernel.org
 L:	ath9k-devel@lists.ath9k.org
 S:	Supported
 
+ATHEROS AR9170 WIRELESS DRIVER
+P:	Christian Lamparter
+M:	chunkeey@web.de
+L:	linux-wireless@vger.kernel.org
+W:	http://wireless.kernel.org/en/users/Drivers/ar9170
+S:	Maintained
+F:	drivers/net/wireless/ar9170/
+
 ATI_REMOTE2 DRIVER
 P:	Ville Syrjala
 M:	syrjala@sci.fi
diff --git a/drivers/net/wireless/Kconfig b/drivers/net/wireless/Kconfig
index 612fffe..8a08235 100644
--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -485,6 +485,7 @@ config MWL8K
 source "drivers/net/wireless/p54/Kconfig"
 source "drivers/net/wireless/ath5k/Kconfig"
 source "drivers/net/wireless/ath9k/Kconfig"
+source "drivers/net/wireless/ar9170/Kconfig"
 source "drivers/net/wireless/ipw2x00/Kconfig"
 source "drivers/net/wireless/iwlwifi/Kconfig"
 source "drivers/net/wireless/hostap/Kconfig"
diff --git a/drivers/net/wireless/Makefile b/drivers/net/wireless/Makefile
index d780487..5e7c9ac 100644
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -57,5 +57,6 @@ obj-$(CONFIG_P54_COMMON)	+= p54/
 
 obj-$(CONFIG_ATH5K)	+= ath5k/
 obj-$(CONFIG_ATH9K)	+= ath9k/
+obj-$(CONFIG_AR9170_COMMON)	+= ar9170/
 
 obj-$(CONFIG_MAC80211_HWSIM)	+= mac80211_hwsim.o
diff --git a/drivers/net/wireless/ar9170/Kconfig b/drivers/net/wireless/ar9170/Kconfig
new file mode 100644
index 0000000..f661187
--- /dev/null
+++ b/drivers/net/wireless/ar9170/Kconfig
@@ -0,0 +1,28 @@
+config AR9170_COMMON
+	tristate "Atheros AR9170 support"
+	depends on WLAN_80211 && MAC80211 && EXPERIMENTAL
+	help
+	  This is common code for AR9170 based devices.
+	  This module does nothing by itself - the USB/(SPI) frontends
+	  also need to be enabled in order to support any devices.
+
+	  Say Y if you have the hardware, or M to build a module called
+	  ar9170common.
+
+config AR9170_USB
+	tristate "Atheros AR9170 USB support"
+	depends on AR9170_COMMON && USB
+	select FW_LOADER
+	help
+	  This is a driver for the Atheros "otus" 802.11n USB devices.
+
+	  These devices require additional firmware (2 files).
+	  For now, these files can be downloaded from here:
+	  http://wireless.kernel.org/en/users/Drivers/ar9170
+
+	  If you choose to build a module, it'll be called ar9170usb.
+
+config AR9170_LEDS
+	bool
+	depends on AR9170_COMMON && MAC80211_LEDS && (LEDS_CLASS = y || LEDS_CLASS = AR9170_COMMON)
+	default y
diff --git a/drivers/net/wireless/ar9170/Makefile b/drivers/net/wireless/ar9170/Makefile
new file mode 100644
index 0000000..59b174d
--- /dev/null
+++ b/drivers/net/wireless/ar9170/Makefile
@@ -0,0 +1,5 @@
+ar9170common-objs += main.o cmd.o mac.o phy.o led.o
+ar9170usb-objs += usb.o
+
+obj-$(CONFIG_AR9170_COMMON) += ar9170common.o
+obj-$(CONFIG_AR9170_USB) += ar9170usb.o
